hibernate-core/src/main/java/org/hibernate/dialect/CockroachDialect.java

```diff
@@ -107,7 +107,7 @@ public class CockroachDialect extends Dialect {
 	private final PostgreSQLDriverKind driverKind;
 
 	public CockroachDialect() {
-		this( DatabaseVersion.make( 19, 2 ) );
+		this( DatabaseVersion.make( 21, 1 ) );
```

Comment:
I think I'd prefer to see 

```suggestion
		this( getMinimumSupportedVersion() );
```
instead, if that is possible with Java. Not exactly sure about the rules for accessing instance methods before the "super-constructor", but if that isn't allowed, maybe this one here would help reducing the possibility of a mismatch:

```suggestion
		this( null );
```

If nothing works, or you don't like any of these suggestions, please introduce a `private static final DatabaseVersion MINIMUM_VERSION = DatabaseVersion.make( 21, 1 )` instead and pass that here:

```suggestion
		this( MINIMUM_VERSION );
```

as well as return that in `getMinimumSupportedVersion`.